SEGV when incorrectly attempting to "from" a pool with zq

Repro is with Zed commit 40639dc, which is associated with the changes in #4808 (cc: @mattnibs).

I mistakenly invoked `zq` instead of `zed query` and happened to bump into this stack trace.

```
$ zq -version
Version: v1.10.0-17-g40639dce

$ zq 'from TestPool | count()'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x223709f]

goroutine 1 [running]:
github.com/brimdata/zed/compiler/data.(*Source).IsLake(...)
	/Users/phil/work/zed/compiler/data/source.go:32
github.com/brimdata/zed/compiler/semantic.(*analyzer).semSource(0xc00041f9c8, {0x284b100?, 0xc000054600?})
	/Users/phil/work/zed/compiler/semantic/op.go:160 +0x79f
github.com/brimdata/zed/compiler/semantic.(*analyzer).semTrunk(0xdb?, {{0xc00075b6d0, 0x5}, {0x284b100, 0xc000054600}, {0x0, 0x0, 0x0}}, {0x0, 0x0, ...})
	/Users/phil/work/zed/compiler/semantic/op.go:66 +0xf1
github.com/brimdata/zed/compiler/semantic.(*analyzer).semFrom(0x24ba820?, 0xc0007334a0?, {0x0?, 0x23f39a0?, 0xc00041f2c0?})
	/Users/phil/work/zed/compiler/semantic/op.go:42 +0x125
github.com/brimdata/zed/compiler/semantic.(*analyzer).semOp(0x60?, {0x284b0e0?, 0xc000770660?}, {0x0?, 0xc00041f960?, 0x1011045?})
	/Users/phil/work/zed/compiler/semantic/op.go:412 +0x79
github.com/brimdata/zed/compiler/semantic.(*analyzer).semSeq(0x100?, {0xc00076c120?, 0x2, 0xc000718fd8?})
	/Users/phil/work/zed/compiler/semantic/op.go:29 +0x85
github.com/brimdata/zed/compiler/semantic.Analyze({0x285f918?, 0x3625e40}, {0xc00076c120, 0x2, 0x2}, 0x0, 0x0)
	/Users/phil/work/zed/compiler/semantic/analyzer.go:19 +0x1db
github.com/brimdata/zed/cli/queryflags.(*Flags).ParseSourcesAndInputs(0xc0001384d8, {0xc000040050, 0x1, 0x1})
	/Users/phil/work/zed/cli/queryflags/flags.go:40 +0x1d9
github.com/brimdata/zed/cli/zq.(*Command).Run(0xc000138380, {0xc000040050, 0x1, 0x1})
	/Users/phil/work/zed/cli/zq/command.go:132 +0x24a
github.com/brimdata/zed/pkg/charm.path.run({0xc000638290?, 0x1, 0x1}, {0xc000040050?, 0x1, 0x0?})
	/Users/phil/work/zed/pkg/charm/path.go:11 +0x79
github.com/brimdata/zed/pkg/charm.(*Spec).ExecRoot(0x100874b?, {0xc000040050, 0x1, 0x1})
	/Users/phil/work/zed/pkg/charm/charm.go:63 +0x3f
main.main()
	/Users/phil/work/zed/cmd/zq/main.go:11 +0x5b
```

I confirmed via binary search that this doesn't happen at the commit right before that.

```
$ zq -version
Version: v1.10.0-16-gef9695f2

$ zq 'from TestPool | count()'
semantic analyzer: from pool cannot be used without a lake
```

--- a/cli/queryflags/flags.go
+++ b/cli/queryflags/flags.go
@@ -11,7 +11,9 @@ import (
 	"github.com/brimdata/zed/cli"
 	"github.com/brimdata/zed/compiler"
 	"github.com/brimdata/zed/compiler/ast"
+	"github.com/brimdata/zed/compiler/data"
 	"github.com/brimdata/zed/compiler/semantic"
+	"github.com/brimdata/zed/pkg/storage"
 	"github.com/brimdata/zed/zbuf"
 	"github.com/brimdata/zed/zson"
 )
@@ -37,7 +39,7 @@ func (f *Flags) ParseSourcesAndInputs(paths []string) ([]string, ast.Seq, bool, 
 			// and appears to start with a from or yield operator.
 			// Otherwise, consider it a file.
 			if query, err := compiler.Parse(src, f.Includes...); err == nil {
-				if s, err := semantic.Analyze(context.Background(), query, nil, nil); err == nil {
+				if s, err := semantic.Analyze(context.Background(), query, data.NewSource(storage.NewLocalEngine(), nil), nil); err == nil {
 					if semantic.HasSource(s) {
 						return nil, query, false, nil
 					}

--- a/cmd/zq/ztests/from-pool-error.yaml
+++ b/cmd/zq/ztests/from-pool-error.yaml
@@ -1,0 +1,7 @@
+script: |
+  ! zq 'from ( pool a )'
+
+outputs:
+  - name: stderr
+    data: |
+      zq: no such file: from ( pool a )

